lib/palindrome_check.rb

```diff
@@ -1,5 +1,30 @@
 # A method to check if the input string is a palindrome.
 # Return true if the string is a palindrome. Return false otherwise.
 def palindrome_check(my_phrase)
-  raise NotImplementedError
+  if my_phrase == nil
+    return false
+  end
+
+  i = 0
+  j = my_phrase.length - 1
+
+  while i <= j
+    while my_phrase[i] == ' '
```

Comment:
Continue to check that `i <= j` before comparing the character at index i to a white space.